service/history/workflow/mutable_state_impl.go

```diff
@@ -2096,11 +2103,13 @@ func (ms *MutableStateImpl) addResetPointFromCompletion(
 	return true
 }
 
-func (ms *MutableStateImpl) updateBuildIdsSearchAttribute(
-	version *commonpb.WorkerVersionStamp,
-	eventID int64,
-	maxSearchAttributeValueSize int,
-) error {
+func (ms *MutableStateImpl) UpdateBuildIdAssignment(buildId string) error {
```

Comment:
what about reset points, i.e. adding a reset point so we can reset to before the wf was moved to this new buildid? if it's not in this PR, can you add a TODO where it has to be changed?

Reply:
this is only used for the first WF task in this PR. I'm not sure if we need to add a reset point for the first task. One redirect rules come to the picture reset points will become relevant. Nevertheless, I think the reset point stuff should go to a separate PR.